ruby/src/IceRuby/Types.h

```diff
@@ -129,7 +132,7 @@ class StreamUtil
 //
 // Base class for type information.
 //
-class TypeInfo : public UnmarshalCallback
+class TypeInfo : public UnmarshalCallback, public std::enable_shared_from_this<TypeInfo>
```

Comment:
Would it make more sense to move this public std::enable_shared_from_this<...> to the derived classes and remove the
```
dynamic_pointer_cast<...>(shared_from_this())
```
?

Reply:
Yes, that makes sense.